### PR TITLE
Added add_ghost_cells to DistributedArray

### PR DESCRIPTION
**Pull request by rohanbabbar04**
- Added method `add_ghost_cells` which adds ghost cells to the local arrays either from the previous or next process.

### CODE REVIEW

**Comment by rohanbabbar04** [comment]
The only issue I faced was if the global shape of x is less than the minimum number of elements(when min_num > 1), especially incase of `_rmatvec_centered3`,` _matvec_centered5` and `_rmatvec_centered5`. To handle these I made `check_matvec_rmatvec` return a boolean and handle that for these three methods only
You can check the method `check_matvec_rmatvec`...
Then i just did 
```
if not self.check_matvec_rmatvec(x, ...):
     # array of zeroes      
     return y
```

**Comment by mrava87** [comment]
@rohanbabbar04 good job!

I am looking at it and writing some comments, but in general I think the approach taken looks good and well implemented

**Review by mrava87 — COMMENTED** [review]
Good start :D 

I pretty much agree with the structure of the code and how you implemented the ghost. I have however to main comments:

- currently the `add_ghost_cells` method seems to assume that the DistributedArray object has been built with `axis=-1`. Am I correct? This is a bit dangerous as `axis=-1` is not even the default parameter. It should be easy to generalize this method?
- similar observation for the FirstDerivative operator.

I think we should fix point1 here, whilst point2 could be tackled with a second PR :)

**Comment by rohanbabbar04** [comment]
One thing @mrava87, for the ghosted cells should we change the whole DistributedArray itself meaning should we create a ghosted DistributedArray(copy of the present array plus the ghosted cells) or we keep it as a ghosted numpy array like I have done?

File: pylops_mpi/DistributedArray.py [hunk 1]
```diff
@@ -424,6 +424,32 @@ def copy(self):
...
+        Parameters
+        ----------
+        cells_front : :obj:`int`, optional
+            Number of cells to be added from the previous process
+            to the start of the array at each rank. Defaults to ``None``.
+        cells_back : :obj:`int`, optional
+            Number of cells to be added from the next process
+            to the back of the array at each rank. Defaults to ``None``.
+        """
+        ghosted_array = self.local_array
```
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply pylops_mpi/DistributedArray.py:440]: Yes to that it remains unchanged.. only `ghosted_array` is populated and returned at each rank...

[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply pylops_mpi/DistributedArray.py:440]: Thanks for pointing this I will change it, currently it changes it....

**Comment by rohanbabbar04** [comment]
Hi @mrava87, like you said this PR now contains only the `add_ghost_cells`
I realized my mistake, I was actually trying to make the FirstDerivative work that I forgot that `add_ghost_cells` was in DistributedArray and should work for n-d DistributedArray as well...
I have made the changes accordingly, once this is merged we can proceed with FirstDerivative..

**Review by mrava87 — APPROVED** [review]
Great, this looks very good. Feel free to merge :)

**Pull request merged**